examples/text-classification/examples/text-classification-ag-news.rs

```diff
@@ -1,14 +1,15 @@
-use burn::optim::{decay::WeightDecayConfig, momentum::MomentumConfig};
-use text_classification::{training::ExperimentConfig, AgNewsDataset};
+#[cfg(feature = "training")]
+fn training() {
+    use burn::optim::{decay::WeightDecayConfig, momentum::MomentumConfig};
+    use text_classification::{training::ExperimentConfig, AgNewsDataset};
 
-type Backend = burn_autodiff::ADBackendDecorator<burn_tch::TchBackend<f32>>;
+    type Backend = burn_autodiff::ADBackendDecorator<burn_tch::TchBackend<burn::tensor::f16>>;
```

Comment:
f32 might be a safe choice for the examples. On mac it's definitely gets NaN loss. Also older Nvidia cards do not support f16. I yet to test on my NVidia GPU.

Reply:
Yeah, but in the other hand I think it's important to demonstrate f16 capabilities, since most big models will be trained witht that. Maybe we should have a feature flag for that!

Reply:
I agree we should demonstrate the capability. Putting the feature flag might be a good way doing it. Also it's more explicit and would make the users aware that they can use f16. We should mention this in the readme, and maybe in the main readme too? (someday in the book too: https://github.com/burn-rs/burn/issues/200) =D